onnx_xla/onnx_xla_interface.cc

```diff
@@ -83,80 +83,80 @@ ONNXIFI_SYMBOL_NAME(onnxReleaseBackendID)(onnxBackendID backendID) {
 ONNXIFI_PUBLIC ONNXIFI_CHECK_RESULT onnxStatus ONNXIFI_ABI ONNXIFI_SYMBOL_NAME(
     onnxGetBackendInfo)(onnxBackendID backendID, onnxBackendInfo infoType,
                         void *infoValue, size_t *infoValueSize) {
-  try {
+  ONNXIFI_TRY_CATCH(([&, backendID, infoType, infoValue, infoValueSize] {
+
+    auto SET_STRING = [&, infoValue, infoValueSize](const char* str)  {
+                        strncpy((char *)(infoValue), str, *infoValueSize);
+                        *infoValueSize = strlen(str) + 1;
+                        return ONNXIFI_STATUS_SUCCESS;
+                      };
+    auto SET_UINT64 = [&, infoValue, infoValueSize](uint64_t x)  {
+                        if (*infoValueSize < sizeof(uint64_t)) {
+                          return ONNXIFI_STATUS_INVALID_POINTER;
+                        }                                
+                        *(uint64_t *)(infoValue) = x;                                              
+                        *infoValueSize = sizeof(uint64_t);
+                        return ONNXIFI_STATUS_SUCCESS;
+                      };
+                                        
     if (!infoValueSize) {
       return ONNXIFI_STATUS_INVALID_POINTER;
     }
-#define SET_STRING(str)                                                        \
-  {                                                                            \
-    snprintf((char *)(infoValue), *infoValueSize, str);                        \
-    *infoValueSize = strlen(str) + 1;                                          \
-  }
-
-#define SET_UINT64(x)                                                          \
-  {                                                                            \
-    if (*infoValueSize < sizeof(uint64_t)) {                                   \
-      return ONNXIFI_STATUS_INVALID_POINTER;                                   \
-    }                                                                          \
-    *(uint64_t *)(infoValue) = x;                                              \
-    *infoValueSize = sizeof(uint64_t);                                         \
-  }
     if (infoType == ONNXIFI_BACKEND_NAME) {
```

Comment:
Any reason you don't use `switch` here?

Reply:
No, I had just copied this from the Tensor-RT PR; changed to switch